### ISSUE TITLE
Implement CAPTCHA spam solution for forms on partner site

### ISSUE THREAD

**Issue by pyrello** [uiowa/uiowa#7394]
# Proposed solution

Build a solution adding a CAPTCHA on forms on a partner website who is experiencing issues with spam.
Work with TM/ISPO to resolve accessibility/security concerns.
Potentially transition the solution from a site split to a feature split.

**Comment by joewhitsitt** [comment]
Got a response from Kirk (in its-web) already saying reCAPTCHA probably won't pass a security review and to start with [hCaptcha](https://www.hcaptcha.com/). https://www.drupal.org/project/hcaptcha

**Comment by mjoneill** [comment]
Question: impact to partners across the river?
(Can discuss in PL)

**Comment by e-marie-w** [comment]
Capturing some knowledge I've gained so far in working on this:
- The free (aka 'publisher') version of hcaptcha supports up to one million requests per month. No info on what happens if you reach that level. Like, can a user no longer successfully submit the webform, or does the hcaptcha simply stop appearing/blocking spam?
- At the free level, users will see the hcaptcha challenge every time they go to submit a webform with it added. For passive mode options we would need a paid version. (Comparisons [here](https://www.hcaptcha.com/pricing) and [here](https://www.hcaptcha.com/plans))
- We can change the challenge level. Options include auto, easy, moderate, difficult. It defaults to moderate when creating the key. 
- The mode setting and challenge level are both at the key level. Multiple keys can be created, if we decided different sites needed different settings. I have not found any information on the maximum number of keys that can be created. 
- As noted elsewhere, accessibility is a concern. According to the [site](https://www.hcaptcha.com/accessibility) they believe they are in 'full Section 508 and WCAG 2.1 AA compliance.' But accessibility users need to jump through the hoop of making an account to automatically pass through captchas and/or receive accessibility-friendly captchas. The site also mentioned text-based captcha challenges can be added, though I'm not seeing that option in the Drupal module.

**Comment by joewhitsitt** [comment]
Responding to a few Slack items:

>the 'apply to all forms' option isn't applying

My guess is that this is a setting for things other than webform. Login block (not used), etc.

>the hcaptcha isn't a required field by default, so spam can just ignore it

Is this because of the admin mode or your admin permissions? I'd be curious if the captcha issue queue had any guidance/issues related to this.

**Comment by joewhitsitt** [comment]
Looks like I might have spoke too soon in EoD. Webform doesn't have a catchall base form id... https://www.drupal.org/project/webform/issues/3378266 so unless that form takes a wildcard like our config.ignore we probably can't just do that.

**Comment by joewhitsitt** [comment]
Based on what I have seen, I wonder if we just go with the editor added/per webform approach. Webforms that are being hard hit or that are probably going to get hit, they can add this to the form.

Otherwise, we might be looking at a hook to set this just for webforms. Actually might be the best way to ensure that multipage forms are still placing it in the right spot. https://git.drupalcode.org/project/captcha/-/blob/2.x/captcha.api.php?ref_type=heads#L186

**Comment by e-marie-w** [comment]
I can switch back to that and see if I can disable the captcha select list part of the element.

**Comment by joewhitsitt** [comment]
We have some previous code to adjust webform components https://github.com/uiowa/uiowa/blob/main/docroot/profiles/custom/sitenow/sitenow.profile#L554
